app/src/main/java/com/pvp/app/ui/theme/Theme.kt

```diff
@@ -27,20 +34,32 @@ private val LightColorScheme = lightColorScheme(
     tertiary = LightTertiary
 )
 
+@HiltViewModel
+class ThemeViewModel @Inject constructor(
+    private val settingService: SettingService
+) : ViewModel() {
+    fun getSetting(): Flow<Int> {
+        return settingService.get(Setting.ApplicationTheme)
+    }
+}
+
 @Composable
 fun CalendarTheme(
+    model: ThemeViewModel,
     darkTheme: Boolean = isSystemInDarkTheme(),
```

Comment:
since this argument is not really used from the outside ever, maybe it could be removed and then the function call could just be moved into the if condition